### DISCUSSION TITLE
Can't Deploy my test script

### DISCUSSION THREAD

**Discussion by Xazrael** [zkSync-Community-Hub/zksync-developers#340] [Dev Tooling] [answered]
![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/20703061/27812cf8-fb67-4847-9bf8-9e9cda6e3e58)

Hello everyone i am new on this things trying to understand how to deploy and write scripts

what is wrong with that script i can't deploy about gas fees i know but i tried to increase gasses and gweis nothing changed it keeps canceling 

![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/20703061/922b4885-ac97-4b6f-91b3-d38bdf217861)

Script is standart script on Remix 

// SPDX-License-Identifier: GPL-3.0

pragma solidity >=0.8.2 <0.9.0;

/**
 * @title Storage
 * @dev Store & retrieve value in a variable
 * @custom:dev-run-script ./scripts/deploy_with_ethers.ts
 */
contract Storage {

    uint256 number;

    /**
     * @dev Store value in variable
     * @param num value to store
     */
    function store(uint256 num) public {
        number = num;
    }

    /**
     * @dev Return value 
     * @return value of 'number'
     */
    function retrieve() public view returns (uint256){
        return number;
    }
}

**Comment by jaguard2021** [comment]
It looks like this is a massage transfer contract.  

![greg](https://github.com/zkSync-Community-Hub/zksync-developers/assets/88972434/b38aee3c-678e-4562-af60-6811d01cd371)

**Comment by 0xhwk** [comment, marked as answer]
As ZkSync is EVM compatible but not equivalent, you can't compile and deploy a contract with the compilers remix is providing. You need zksync-solc to compile the contracts you wish to deploy. Docs recommend using hardhat to compile and deploy. Also people recommend using https://app.atlaszk.com/ide as it has the compiler, but I have never tried it.

**Reply by jaguard2021** [reply]
Using Atlas it is much easier to create programs in ZkSync. I have tried it and it works, and the contract verification process is easy

**Comment by jecgaegacharsmi** [comment]
Your Solidity script appears to be correct and standard. Gas fees and gas limits are common issues when deploying contracts. Ensure your gas limit is sufficient for deployment and consider adjusting the gas price (Gwei) to ensure the transaction gets processed promptly. Additionally, verify that you're deploying to a compatible network and that your account has enough funds to cover the deployment costs.

**Comment by hunastly** [comment]
Your contract code looks correct. Gas fees might be high due to network congestion. Try deploying during off-peak hours or increasing gas limit and gas price further.

**Comment by AlbionaHoti** [comment]
Hi @Xazrael, are you still having issues deploying the contract on the Remix IDE? [Here's the guide to help you get started](https://medium.com/nethermind-eth/the-zksync-era-remix-plugin-a-how-to-guide-fc54e8d24bd3).

**Comment by bxpana** [comment]
@0xhwk had a question regarding potential moderating. Hit me up at support@zksync.io or let me know what's the best way to contact you 😁
